### ISSUE TITLE
FileNotFoundError: [Errno 2] No such file or directory: 'lsb_release' when running `idaes get-extensions` in containerized image

### ISSUE THREAD

**Issue by lbianchi-lbl** [IDAES/idaes-pse#1221]
- Experienced when trying to set up Binder (IDAES/examples#61)
- The workaround is to specify the distro explicitly (`idaes get-extensions --distro ubuntu1804` for Binder)
- It could be argued that this should be handled more gracefully on the side of Pyomo's `pyomo.common.download.FileDownloader`, in which case we might want to add this as a possible enhancement (I'm not sure if this can be considered a bug)

```
Traceback (most recent call last):
  File "/srv/conda/envs/notebook/bin/idaes", line 8, in <module>
    sys.exit(command_base())
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/idaes/commands/extensions.py", line 128, in get_extensions
    d = idaes.commands.util.download_bin.download_binaries(
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/idaes/commands/util/download_bin.py", line 364, in download_binaries
    arch, platform = _get_arch_and_platform(  # pylint: disable=unused-variable
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/idaes/commands/util/download_bin.py", line 74, in _get_arch_and_platform
    platform = fd.get_os_version().replace(".", "")
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/pyomo/common/download.py", line 199, in get_os_version
    FileDownloader._os_version = cls._get_os_version()
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/pyomo/common/download.py", line 156, in _get_os_version
    subprocess.run(
  File "/srv/conda/envs/notebook/lib/python3.10/subprocess.py", line 503, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/srv/conda/envs/notebook/lib/python3.10/subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/srv/conda/envs/notebook/lib/python3.10/subprocess.py", line 1863, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'lsb_release'
```

**Comment by lbianchi-lbl** [comment]
@blnicho suggests to open an issue in Pyomo once we managed to find steps to reproduce it; @eslickj thinks there might already be an issue from when he encountered the same (or similar) error.

**Comment by jsiirola** [comment]
See if the following resolves the error:
```patch
diff --git a/pyomo/common/download.py b/pyomo/common/download.py
index a3d5b3e..79d5302 100644
--- a/pyomo/common/download.py
+++ b/pyomo/common/download.py
@@ -15,6 +15,7 @@ import logging
 import os
 import platform
 import re
+import shutil
 import sys
 import subprocess

@@ -99,14 +100,15 @@ class FileDownloader(object):

     @classmethod
     def _get_distver_from_lsb_release(cls):
+        lsb_release = shutil.which('lsb_release')
         dist = subprocess.run(
-            ['lsb_release', '-si'],
+            [lsb_release, '-si'],
             stdout=subprocess.PIPE,
             stderr=subprocess.STDOUT,
             universal_newlines=True,
         )
         ver = subprocess.run(
-            ['lsb_release', '-sr'],
+            [lsb_release, '-sr'],
             stdout=subprocess.PIPE,
             stderr=subprocess.STDOUT,
             universal_newlines=True,
@@ -152,14 +154,7 @@ class FileDownloader(object):
                 dist, ver = cls._get_distver_from_distro()
             elif os.path.exists('/etc/redhat-release'):
                 dist, ver = cls._get_distver_from_redhat_release()
-            elif (
-                subprocess.run(
-                    ['lsb_release'],
-                    stdout=subprocess.DEVNULL,
-                    stderr=subprocess.DEVNULL,
-                ).returncode
-                == 0
-            ):
+            elif shutil.which('lsb_release'):
                 dist, ver = cls._get_distver_from_lsb_release()
             elif os.path.exists('/etc/os-release'):
                 # Note that (at least on centos), os_release is an
```

...I will get a PR up to Pyomo shortly

**Comment by ksbeattie** [comment]
This should be fixed by Pyomo 6.6.2
